### PR TITLE
!!! TASK: Add type declarations to persistence interfaces

### PR DESCRIPTION
**Pull request by kdambekalns**

### CODE REVIEW

File: Neos.Flow/Classes/Persistence/Doctrine/Query.php [hunk 1]
```diff
@@ -69,7 +68,7 @@ class Query implements QueryInterface
     /**
      * @var array
      */
-    protected $orderings;
+    protected $orderings = [];
 
     /**
      * @var integer
```
Comment (albe) [inline Neos.Flow/Classes/Persistence/Doctrine/Query.php:74]:
```suggestion
     * @var integer|null
```
Same for the offset below, but GH doesn't allow suggestions/comments on lines outside the diff lines

**Review by albe — COMMENTED** [review]
Not sure about the ```ERROR: ImplementedReturnTypeMismatch - Packages/Framework/Neos.Flow/Classes/Persistence/Doctrine/Repository.php:130:16 - The inherited return type 'array<array-key, object>' for Doctrine\Persistence\ObjectRepository::findAll is different to the implemented return type for Neos\Flow\Persistence\Doctrine\Repository::findall 'Neos\Flow\Persistence\QueryResultInterface' (see https://psalm.dev/123)
     * @return QueryResultInterface The query result```
But guess we need to baseline this.

**Comment by kdambekalns** [comment]
> `The inherited return type 'array<array-key, object>' for Doctrine\Persistence\ObjectRepository::findAll is different to the implemented return type for Neos\Flow\Persistence\Doctrine\Repository::findall 'Neos\Flow\Persistence\QueryResultInterface'`
> But guess we need to baseline this.

Yeah. The `QueryResultInterface` extends `\Countable, \Iterator, \ArrayAccess`, so it fits, even though strrictly speaking is breaks the contract. But it's only a "docblock contract", no real return type is declared in Doctrine.

Should I baseline this within this PR?

**Comment by albe** [comment]
Yeah, would be good I think.

**Comment by albe** [comment]
green ✔️

**Review by albe — APPROVED** [review]

**Pull request merged**